# Ultralytics Framework for YOLO11 Training

In [1]:
import torch

# Number of GPUs available
num_gpus = torch.cuda.device_count()

print(f"Number of available GPUs: {num_gpus}")


Number of available GPUs: 4


# Yolo11n Model Training on:
- Carla_Labelling-v5 dataset
- Train: 70% = 4485 Images
- Validation: 20% = 1274 Images
- Test: 10% = 641 Images

- Hyperparameters are on automatic settings using Ultralytics package

In [2]:
from ultralytics import YOLO

# Load YOLOv10n model from scratch
model = YOLO("yolo11n.pt")

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 


100%|██████████| 5.35M/5.35M [00:00<00:00, 99.0MB/s]


# Validate the pretrained weights by running inference on an image.

In [3]:
# Run batched inference on a list of images
results = model(["/home/sur06423/project/1.jpg"])  # return a list of Results objects

# Process results list
for result in results:
    boxes = result.boxes  # Boxes object for bounding box outputs
    masks = result.masks  # Masks object for segmentation masks outputs
    keypoints = result.keypoints  # Keypoints object for pose outputs
    probs = result.probs  # Probs object for classification outputs
    obb = result.obb  # Oriented boxes object for OBB outputs
    result.show()  # display to screen
    result.save(filename="result.jpg")  # save to disk


0: 448x640 4 horses, 69.5ms
Speed: 6.6ms preprocess, 69.5ms inference, 509.3ms postprocess per image at shape (1, 3, 448, 640)


Error: no "view" rule for type "image/png" passed its test case
       (for more information, add "--debug=1" on the command line)


# Train the model using the following guidelines:
- Link for various arguments: https://docs.ultralytics.com/modes/train/#train-settings 

In [4]:
import os

# Set library paths
os.environ['LD_LIBRARY_PATH'] = '/usr/lib/xorg-nvidia-525.116.04/lib/x86_64-linux-gnu:' + os.environ.get('LD_LIBRARY_PATH', '')
os.environ['LD_LIBRARY_PATH'] = '/usr/lib/xorg/lib/x86_64-linux-gnu:' + os.environ['LD_LIBRARY_PATH']
os.environ['LD_LIBRARY_PATH'] = '/usr/lib/xorg-nvidia-535.113.01/lib/x86_64-linux-gnu:' + os.environ['LD_LIBRARY_PATH']

# Verify the update
print(os.environ['LD_LIBRARY_PATH'])


/usr/lib/xorg-nvidia-535.113.01/lib/x86_64-linux-gnu:/usr/lib/xorg/lib/x86_64-linux-gnu:/usr/lib/xorg-nvidia-525.116.04/lib/x86_64-linux-gnu:/home/sur06423/miniconda3/envs/yolov_env/lib/python3.9/site-packages/cv2/../../lib64:


In [5]:
# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
model.train(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", # 	Path to the dataset configuration file (e.g., coco8.yaml). This file contains dataset-specific parameters, including paths to training and validation data, class names, and number of classes.
            epochs=100, # Total number of training epochs. Each epoch represents a full pass over the entire dataset. Adjusting this value can affect training duration and model performance.
            patience=10, # Number of epochs to wait without improvement in validation metrics before early stopping the training. Helps prevent overfitting by stopping training when performance plateaus.
            batch=128, # Batch size, with three modes: set as an integer (e.g., batch=16), auto mode for 60% GPU memory utilization (batch=-1), or auto mode with specified utilization fraction (batch=0.70).
            imgsz=640, # Target image size for training. All images are resized to this dimension before being fed into the model. Affects model accuracy and computational complexity.
            save=True, # Enables saving of training checkpoints and final model weights. Useful for resuming training or model deployment.
            save_period = 1, # Frequency of saving model checkpoints, specified in epochs. A value of -1 disables this feature. Useful for saving interim models during long training sessions.
            device=[0,1,2,3], # Specifies the computational device(s) for training: a single GPU (device=0), multiple GPUs (device=0,1), CPU (device=cpu), or MPS for Apple silicon (device=mps).
            workers=6, # Number of worker threads for data loading (per RANK if Multi-GPU training). Influences the speed of data preprocessing and feeding into the model, especially useful in multi-GPU setups.
            project="/home/sur06423/project/Version_5_exp/YOLO_11/v11_n", # Name of the project directory where training outputs are saved. Allows for organized storage of different experiments.
            pretrained = True, # Determines whether to start training from a pretrained model. Can be a boolean value or a string path to a specific model from which to load weights. Enhances training efficiency and model performance.
            optimizer = "auto", # Choice of optimizer for training. Options include SGD, Adam, AdamW, NAdam, RAdam, RMSProp etc., or auto for automatic selection based on model configuration. Affects convergence speed and stability.
            verbose = True, # Enables verbose output during training, providing detailed logs and progress updates. Useful for debugging and closely monitoring the training process.
            seed = 0, # Sets the random seed for training, ensuring reproducibility of results across runs with the same configurations.
            amp = True, # Enables Automatic Mixed Precision (AMP) training, reducing memory usage and possibly speeding up training with minimal impact on accuracy.
            profile =True, # Enables profiling of ONNX and TensorRT speeds during training, useful for optimizing model deployment.
            val=True, # Enables validation during training, allowing for periodic evaluation of model performance on a separate dataset.
            plots = True, #Generates and saves plots of training and validation metrics, as well as prediction examples, providing visual insights into model performance and learning progression.                
)

New https://pypi.org/project/ultralytics/8.3.14 available 😃 Update with 'pip install -U ultralytics'
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
                                                      CUDA:1 (NVIDIA RTX A6000, 48677MiB)
                                                      CUDA:2 (NVIDIA RTX A6000, 48677MiB)
                                                      CUDA:3 (NVIDIA RTX A6000, 48677MiB)
engine/trainer: task=detect, mode=train, model=yolo11n.pt, data=/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml, epochs=100, time=None, patience=10, batch=128, imgsz=640, save=True, save_period=1, cache=False, device=[0, 1, 2, 3], workers=6, project=/home/sur06423/project/Version_5_exp/YOLO_11/v11_n, name=train, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=True, freeze

Overriding model.yaml nc=80 with nc=4

                   from  n    params  module                                       arguments                     
  0                  -1  1       464  ultralytics.nn.modules.conv.Conv             [3, 16, 3, 2]                 
  1                  -1  1      4672  ultralytics.nn.modules.conv.Conv             [16, 32, 3, 2]                
  2                  -1  1      6640  ultralytics.nn.modules.block.C3k2            [32, 64, 1, False, 0.25]      
  3                  -1  1     36992  ultralytics.nn.modules.conv.Conv             [64, 64, 3, 2]                
  4                  -1  1     26080  ultralytics.nn.modules.block.C3k2            [64, 128, 1, False, 0.25]     
  5                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]              
  6                  -1  1     87040  ultralytics.nn.modules.block.C3k2            [128, 128, 1, True]           
  7                  -1  1    295424  ultralytics

train: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/train/labels.cache... 4485 images, 430 backgrounds, 0 corrupt: 100%|██████████| 4485/4485 [00:00<?, ?it/s]
val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/valid/labels.cache... 1274 images, 124 backgrounds, 0 corrupt: 100%|██████████| 1274/1274 [00:00<?, ?it/s]


Plotting labels to /home/sur06423/project/Version_5_exp/YOLO_11/v11_n/train/labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.000714, momentum=0.9) with parameter groups 81 weight(decay=0.0), 88 weight(decay=0.001), 87 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 24 dataloader workers
Logging results to /home/sur06423/project/Version_5_exp/YOLO_11/v11_n/train
Starting training for 100 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      1/100      4.85G      1.177       3.33      1.092          5        640: 100%|██████████| 36/36 [00:10<00:00,  3.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.83it/s]


                   all       1274       7377       0.01      0.195      0.104     0.0837

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      2/100      4.92G      1.039      1.528      1.002         17        640: 100%|██████████| 36/36 [00:08<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.37it/s]


                   all       1274       7377      0.996     0.0453      0.198      0.143

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      3/100      4.79G      1.002      1.257      1.004         37        640: 100%|██████████| 36/36 [00:08<00:00,  4.47it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.31it/s]


                   all       1274       7377      0.828      0.168      0.278      0.175

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      4/100      4.85G     0.9735      1.111      1.001          8        640: 100%|██████████| 36/36 [00:07<00:00,  4.53it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.42it/s]


                   all       1274       7377      0.688      0.306      0.353      0.217

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      5/100      4.85G      0.958     0.9751     0.9856          6        640: 100%|██████████| 36/36 [00:07<00:00,  4.51it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.64it/s]


                   all       1274       7377      0.733      0.369      0.417      0.255

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      6/100      4.82G     0.9241     0.8577     0.9684         26        640: 100%|██████████| 36/36 [00:07<00:00,  4.60it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.79it/s]


                   all       1274       7377      0.719      0.361      0.407       0.25

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      7/100      5.08G      0.913     0.7997     0.9655          3        640: 100%|██████████| 36/36 [00:07<00:00,  4.63it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.89it/s]


                   all       1274       7377      0.727      0.415      0.468      0.276

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      8/100      4.86G     0.9082     0.7597      0.951         24        640: 100%|██████████| 36/36 [00:07<00:00,  4.50it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.86it/s]


                   all       1274       7377      0.754       0.43      0.492      0.292

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      9/100      4.89G     0.8764     0.7025     0.9584          1        640: 100%|██████████| 36/36 [00:08<00:00,  4.47it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.84it/s]


                   all       1274       7377      0.757      0.426      0.489      0.304

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     10/100      4.78G     0.8857     0.6944     0.9456          8        640: 100%|██████████| 36/36 [00:08<00:00,  4.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.75it/s]


                   all       1274       7377      0.758      0.453      0.507      0.302

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     11/100      4.85G     0.9034     0.6781     0.9409         14        640: 100%|██████████| 36/36 [00:08<00:00,  4.14it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.83it/s]


                   all       1274       7377      0.722      0.464      0.512      0.307

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     12/100      4.86G     0.8929     0.7956     0.9211          1        640: 100%|██████████| 36/36 [00:07<00:00,  4.61it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.91it/s]


                   all       1274       7377      0.782      0.461      0.528      0.307

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     13/100      4.93G      0.874      0.639     0.9411         11        640: 100%|██████████| 36/36 [00:07<00:00,  4.51it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.93it/s]


                   all       1274       7377      0.773      0.455      0.509      0.311

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     14/100      4.98G     0.8744      0.664     0.9352         14        640: 100%|██████████| 36/36 [00:07<00:00,  4.60it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.84it/s]


                   all       1274       7377      0.823      0.454      0.523      0.326

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     15/100      4.84G     0.8537     0.6124     0.9305         24        640: 100%|██████████| 36/36 [00:07<00:00,  4.56it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.98it/s]


                   all       1274       7377      0.792      0.491      0.557      0.331

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     16/100      4.89G     0.8414     0.6071     0.9297          4        640: 100%|██████████| 36/36 [00:08<00:00,  4.49it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.57it/s]


                   all       1274       7377      0.795      0.452      0.525      0.329

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     17/100      4.92G     0.8554     0.6125     0.9509         15        640: 100%|██████████| 36/36 [00:07<00:00,  4.55it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.92it/s]


                   all       1274       7377      0.779      0.488      0.549      0.338

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     18/100      4.96G     0.8288     0.6113     0.9409          6        640: 100%|██████████| 36/36 [00:07<00:00,  4.58it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.85it/s]


                   all       1274       7377      0.789      0.494      0.555      0.346

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     19/100      5.14G     0.8559     0.5835      0.928         12        640: 100%|██████████| 36/36 [00:07<00:00,  4.60it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.98it/s]


                   all       1274       7377      0.792      0.492      0.553      0.339

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     20/100      5.07G     0.8377     0.5791     0.9297         23        640: 100%|██████████| 36/36 [00:07<00:00,  4.53it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.88it/s]


                   all       1274       7377      0.784      0.506      0.561      0.347

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     21/100       4.9G     0.8282     0.5698     0.9276          4        640: 100%|██████████| 36/36 [00:07<00:00,  4.55it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.97it/s]


                   all       1274       7377      0.806      0.486      0.556      0.347

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     22/100      4.93G     0.8481     0.5967     0.9249          3        640: 100%|██████████| 36/36 [00:07<00:00,  4.60it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.53it/s]


                   all       1274       7377      0.814      0.495      0.566      0.354

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     23/100      4.86G     0.7976     0.5499     0.9149         27        640: 100%|██████████| 36/36 [00:08<00:00,  4.49it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.97it/s]


                   all       1274       7377      0.821      0.501       0.57      0.351

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     24/100      4.85G     0.8087     0.5409     0.9165          3        640: 100%|██████████| 36/36 [00:07<00:00,  4.60it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.87it/s]


                   all       1274       7377       0.81      0.511      0.575      0.354

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     25/100      4.86G     0.8058     0.5485     0.9174         20        640: 100%|██████████| 36/36 [00:07<00:00,  4.57it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.78it/s]


                   all       1274       7377      0.794        0.5      0.572      0.353

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     26/100      4.78G     0.7894     0.5221     0.9285         18        640: 100%|██████████| 36/36 [00:07<00:00,  4.63it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.94it/s]


                   all       1274       7377      0.782      0.489      0.557      0.349

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     27/100      5.05G     0.8389     0.5436     0.9183         11        640: 100%|██████████| 36/36 [00:07<00:00,  4.71it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.99it/s]


                   all       1274       7377        0.8      0.496      0.571      0.359

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     28/100      4.79G     0.7893     0.5304     0.9131         20        640: 100%|██████████| 36/36 [00:08<00:00,  4.50it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.92it/s]


                   all       1274       7377      0.779      0.519      0.576      0.358

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     29/100      4.93G     0.7698     0.5104     0.9033          7        640: 100%|██████████| 36/36 [00:08<00:00,  4.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.81it/s]


                   all       1274       7377      0.815      0.516      0.582       0.36

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     30/100      4.86G     0.8026     0.5251     0.9177         11        640: 100%|██████████| 36/36 [00:07<00:00,  4.55it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.91it/s]


                   all       1274       7377      0.835      0.514      0.587      0.368

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     31/100      4.92G     0.7809     0.5155     0.9177         19        640: 100%|██████████| 36/36 [00:07<00:00,  4.62it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.95it/s]


                   all       1274       7377      0.829      0.519      0.601      0.371

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     32/100      5.06G     0.7814     0.5126     0.9043         10        640: 100%|██████████| 36/36 [00:07<00:00,  4.53it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.98it/s]


                   all       1274       7377      0.807      0.521      0.592       0.37

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     33/100      5.04G     0.7664     0.5093     0.9066         13        640: 100%|██████████| 36/36 [00:07<00:00,  4.52it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.84it/s]


                   all       1274       7377       0.83      0.535      0.603      0.375

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     34/100      4.88G     0.7769     0.5004     0.9117         11        640: 100%|██████████| 36/36 [00:08<00:00,  4.42it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.95it/s]


                   all       1274       7377      0.822      0.533      0.595      0.371

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     35/100      5.08G     0.7943     0.5075     0.9199         21        640: 100%|██████████| 36/36 [00:08<00:00,  4.47it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.87it/s]


                   all       1274       7377      0.847      0.528      0.602      0.379

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     36/100      4.73G     0.7711     0.5062     0.9082          9        640: 100%|██████████| 36/36 [00:08<00:00,  4.41it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.80it/s]


                   all       1274       7377      0.818      0.533        0.6      0.376

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     37/100      4.79G     0.7722     0.5008      0.911         10        640: 100%|██████████| 36/36 [00:08<00:00,  4.36it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.80it/s]


                   all       1274       7377      0.796      0.544      0.601      0.375

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     38/100      4.83G     0.7323     0.5158      0.889          0        640: 100%|██████████| 36/36 [00:08<00:00,  4.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.02it/s]


                   all       1274       7377      0.828      0.512      0.592       0.37

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     39/100      4.88G     0.7568     0.4805     0.9035         21        640: 100%|██████████| 36/36 [00:07<00:00,  4.56it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.29it/s]


                   all       1274       7377      0.838      0.545      0.611      0.382

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     40/100      4.98G     0.7629     0.4972     0.9017         15        640: 100%|██████████| 36/36 [00:07<00:00,  4.57it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.94it/s]


                   all       1274       7377      0.834      0.545      0.615      0.384

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     41/100      4.98G     0.7429     0.4834     0.9045         10        640: 100%|██████████| 36/36 [00:08<00:00,  4.38it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.99it/s]


                   all       1274       7377       0.82      0.551      0.621      0.387

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     42/100      4.86G     0.7487     0.4798     0.9018         24        640: 100%|██████████| 36/36 [00:08<00:00,  4.36it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.84it/s]


                   all       1274       7377      0.827      0.551      0.615      0.386

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     43/100      4.86G     0.7426     0.4807     0.9103          4        640: 100%|██████████| 36/36 [00:08<00:00,  4.47it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.94it/s]


                   all       1274       7377      0.827      0.553      0.615      0.391

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     44/100      4.98G     0.7577     0.4751     0.8921         26        640: 100%|██████████| 36/36 [00:08<00:00,  4.38it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.94it/s]


                   all       1274       7377      0.824      0.548      0.616      0.386

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     45/100      4.85G     0.7373     0.4733     0.8963          8        640: 100%|██████████| 36/36 [00:08<00:00,  4.42it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.89it/s]


                   all       1274       7377      0.808      0.554      0.606      0.379

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     46/100      4.73G     0.7147      0.457      0.895          8        640: 100%|██████████| 36/36 [00:07<00:00,  4.73it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.01it/s]


                   all       1274       7377       0.84      0.546       0.63      0.397

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     47/100      5.03G      0.727     0.4654     0.8929         15        640: 100%|██████████| 36/36 [00:07<00:00,  4.55it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.97it/s]


                   all       1274       7377      0.828      0.566      0.631      0.398

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     48/100       4.8G       0.74     0.4622     0.8931         16        640: 100%|██████████| 36/36 [00:07<00:00,  4.59it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.95it/s]


                   all       1274       7377      0.831      0.564       0.63      0.396

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     49/100      4.92G     0.7513     0.5068     0.8768          7        640: 100%|██████████| 36/36 [00:07<00:00,  4.69it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.96it/s]


                   all       1274       7377      0.848      0.556       0.63      0.397

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     50/100      4.97G     0.7468     0.4608     0.8932         17        640: 100%|██████████| 36/36 [00:08<00:00,  4.49it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.81it/s]


                   all       1274       7377      0.825       0.55       0.62      0.391

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     51/100       4.9G     0.7326     0.4576     0.9017         10        640: 100%|██████████| 36/36 [00:07<00:00,  4.62it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.97it/s]


                   all       1274       7377      0.819       0.57      0.633      0.398

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     52/100      4.86G     0.7174     0.4504     0.8948          5        640: 100%|██████████| 36/36 [00:07<00:00,  4.66it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.96it/s]


                   all       1274       7377      0.865      0.552      0.635      0.395

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     53/100      4.86G      0.719     0.4585     0.8994          8        640: 100%|██████████| 36/36 [00:07<00:00,  4.63it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.10it/s]


                   all       1274       7377      0.848      0.559      0.639      0.406

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     54/100       4.8G     0.6921     0.4372     0.8859         14        640: 100%|██████████| 36/36 [00:08<00:00,  4.40it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.96it/s]


                   all       1274       7377      0.827      0.559       0.63      0.398

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     55/100      4.74G     0.7244     0.4591     0.8926         21        640: 100%|██████████| 36/36 [00:07<00:00,  4.54it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.68it/s]


                   all       1274       7377       0.85      0.562      0.639      0.407

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     56/100       4.7G     0.6993     0.4446     0.8903          5        640: 100%|██████████| 36/36 [00:07<00:00,  4.53it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.98it/s]


                   all       1274       7377      0.833      0.553      0.631      0.401

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     57/100       4.9G      0.714     0.4454     0.8854         13        640: 100%|██████████| 36/36 [00:07<00:00,  4.54it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.95it/s]


                   all       1274       7377      0.823      0.565      0.637      0.401

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     58/100      4.71G     0.6992     0.4377     0.8918         21        640: 100%|██████████| 36/36 [00:07<00:00,  4.55it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.71it/s]


                   all       1274       7377       0.84      0.574      0.642      0.409

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     59/100      4.84G     0.7121     0.4308     0.8871         12        640: 100%|██████████| 36/36 [00:08<00:00,  4.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.08it/s]


                   all       1274       7377      0.848      0.574       0.65      0.412

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     60/100      4.81G     0.7016     0.4388     0.8913         18        640: 100%|██████████| 36/36 [00:08<00:00,  4.26it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.91it/s]


                   all       1274       7377       0.84       0.57      0.645      0.409

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     61/100      5.01G     0.7142      0.443     0.8876         39        640: 100%|██████████| 36/36 [00:07<00:00,  4.57it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.91it/s]


                   all       1274       7377      0.871      0.571      0.651      0.412

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     62/100      5.04G     0.6868     0.4263     0.8846         33        640: 100%|██████████| 36/36 [00:07<00:00,  4.50it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.97it/s]


                   all       1274       7377      0.855      0.574      0.644      0.411

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     63/100      4.81G     0.7071     0.4384     0.8832         13        640: 100%|██████████| 36/36 [00:07<00:00,  4.68it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.91it/s]


                   all       1274       7377      0.854      0.576      0.648      0.412

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     64/100       4.7G     0.7247     0.4397     0.8844          6        640: 100%|██████████| 36/36 [00:08<00:00,  4.46it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.85it/s]


                   all       1274       7377      0.858      0.572      0.651      0.416

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     65/100      4.88G      0.714     0.4342     0.8853          3        640: 100%|██████████| 36/36 [00:08<00:00,  4.28it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.82it/s]


                   all       1274       7377      0.837       0.58      0.654      0.415

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     66/100      4.84G     0.6973     0.4258     0.8905         10        640: 100%|██████████| 36/36 [00:08<00:00,  4.37it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.98it/s]


                   all       1274       7377      0.844      0.579      0.653      0.417

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     67/100      4.88G     0.7151     0.4382       0.89         19        640: 100%|██████████| 36/36 [00:08<00:00,  4.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.93it/s]


                   all       1274       7377       0.87      0.566      0.649      0.414

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     68/100      5.04G     0.6856     0.4199     0.8782         19        640: 100%|██████████| 36/36 [00:08<00:00,  4.35it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.94it/s]


                   all       1274       7377      0.867      0.573      0.653      0.416

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     69/100      4.98G     0.7119      0.431     0.8871         16        640: 100%|██████████| 36/36 [00:08<00:00,  4.40it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.97it/s]


                   all       1274       7377      0.859      0.578      0.657      0.421

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     70/100       4.8G     0.6941     0.4249     0.8834         24        640: 100%|██████████| 36/36 [00:07<00:00,  4.51it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.57it/s]


                   all       1274       7377      0.837      0.576      0.653       0.42

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     71/100      4.93G     0.6903      0.411     0.8832         17        640: 100%|██████████| 36/36 [00:08<00:00,  4.47it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.94it/s]


                   all       1274       7377      0.832       0.58      0.654       0.42

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     72/100      4.84G     0.6883     0.4093     0.8811         12        640: 100%|██████████| 36/36 [00:07<00:00,  4.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.99it/s]


                   all       1274       7377      0.862      0.569      0.649      0.413

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     73/100      4.83G     0.6778     0.4147     0.8806         13        640: 100%|██████████| 36/36 [00:07<00:00,  4.63it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.16it/s]


                   all       1274       7377      0.873      0.572      0.655      0.419

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     74/100      4.79G     0.6739     0.4044     0.8828         11        640: 100%|██████████| 36/36 [00:08<00:00,  4.20it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.09it/s]


                   all       1274       7377      0.843      0.581      0.657      0.421

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     75/100      4.88G     0.6809      0.412     0.8797          6        640: 100%|██████████| 36/36 [00:07<00:00,  4.62it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.45it/s]


                   all       1274       7377      0.861      0.578      0.655      0.421

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     76/100      4.89G     0.6869     0.4109     0.8726          8        640: 100%|██████████| 36/36 [00:07<00:00,  4.74it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.01it/s]


                   all       1274       7377      0.866      0.578      0.663      0.424

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     77/100      4.84G     0.6896     0.4126      0.878         22        640: 100%|██████████| 36/36 [00:07<00:00,  4.51it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.04it/s]


                   all       1274       7377      0.874      0.572      0.658      0.421

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     78/100      5.06G     0.6804     0.4016     0.8905          4        640: 100%|██████████| 36/36 [00:07<00:00,  4.57it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.03it/s]


                   all       1274       7377      0.842      0.589      0.663      0.425

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     79/100      4.87G     0.6838     0.4068     0.8822         13        640: 100%|██████████| 36/36 [00:07<00:00,  4.69it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.18it/s]


                   all       1274       7377      0.853      0.585      0.666      0.428

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     80/100      5.06G     0.6732     0.4103     0.8779         36        640: 100%|██████████| 36/36 [00:08<00:00,  4.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.92it/s]


                   all       1274       7377      0.856      0.576      0.661       0.42

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     81/100      4.87G     0.6692     0.4084     0.8759          7        640: 100%|██████████| 36/36 [00:07<00:00,  4.56it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.21it/s]


                   all       1274       7377      0.869      0.589       0.67      0.426

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     82/100      4.96G     0.6823      0.414     0.8855          9        640: 100%|██████████| 36/36 [00:07<00:00,  4.62it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.50it/s]


                   all       1274       7377      0.844      0.593      0.668       0.43

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     83/100      4.88G     0.6799     0.4032     0.8778          8        640: 100%|██████████| 36/36 [00:07<00:00,  4.61it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.99it/s]


                   all       1274       7377      0.854       0.59      0.669      0.426

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     84/100      4.83G     0.6737     0.4077     0.8787          8        640: 100%|██████████| 36/36 [00:07<00:00,  4.65it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.75it/s]


                   all       1274       7377      0.879      0.576      0.672      0.431

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     85/100      5.05G     0.6735     0.4011     0.8865         11        640: 100%|██████████| 36/36 [00:07<00:00,  4.60it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.92it/s]


                   all       1274       7377      0.851      0.597      0.674       0.43

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     86/100      4.92G     0.7055     0.4734     0.9395          3        640: 100%|██████████| 36/36 [00:08<00:00,  4.42it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.03it/s]


                   all       1274       7377      0.864      0.588      0.669      0.428

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     87/100      4.75G     0.6757     0.4012     0.8793         14        640: 100%|██████████| 36/36 [00:08<00:00,  4.49it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.99it/s]


                   all       1274       7377      0.854      0.592      0.672      0.432

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     88/100       4.9G     0.6709     0.3971     0.8731         16        640: 100%|██████████| 36/36 [00:08<00:00,  4.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.10it/s]


                   all       1274       7377      0.862      0.589      0.676      0.434

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     89/100      5.05G     0.6375      0.377     0.8774         14        640: 100%|██████████| 36/36 [00:07<00:00,  4.62it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.94it/s]


                   all       1274       7377      0.845      0.601      0.677      0.434

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     90/100      4.71G     0.6469     0.3894     0.8836          3        640: 100%|██████████| 36/36 [00:08<00:00,  4.46it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.02it/s]


                   all       1274       7377      0.875      0.586      0.675      0.435
Closing dataloader mosaic

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     91/100      4.72G     0.7254     0.4069     0.8631         13        640: 100%|██████████| 36/36 [00:08<00:00,  4.07it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.89it/s]


                   all       1274       7377      0.852      0.597      0.673      0.429

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     92/100       4.7G     0.7196     0.3966     0.8656         13        640: 100%|██████████| 36/36 [00:07<00:00,  4.77it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.01it/s]


                   all       1274       7377       0.84      0.595      0.671      0.433

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     93/100      4.77G     0.7259      0.397     0.8617          7        640: 100%|██████████| 36/36 [00:07<00:00,  4.80it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.71it/s]


                   all       1274       7377      0.861      0.596      0.675      0.433

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     94/100      4.89G     0.7236     0.4034     0.8631          6        640: 100%|██████████| 36/36 [00:07<00:00,  4.56it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.95it/s]


                   all       1274       7377      0.852      0.599      0.675      0.434

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     95/100      4.76G     0.7287     0.4058     0.8584         31        640: 100%|██████████| 36/36 [00:07<00:00,  4.72it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.89it/s]


                   all       1274       7377      0.867      0.594      0.676      0.436

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     96/100      4.78G     0.7158     0.4155     0.8644         13        640: 100%|██████████| 36/36 [00:07<00:00,  4.85it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.05it/s]


                   all       1274       7377      0.866      0.592      0.678      0.437

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     97/100       4.8G     0.7219     0.3985     0.8625         13        640: 100%|██████████| 36/36 [00:07<00:00,  4.56it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.89it/s]


                   all       1274       7377      0.863      0.594      0.677      0.436

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     98/100      4.77G     0.7219     0.4006     0.8531          3        640: 100%|██████████| 36/36 [00:07<00:00,  4.59it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.97it/s]


                   all       1274       7377      0.871      0.596      0.678      0.437

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     99/100      4.73G     0.7064     0.3865     0.8554         40        640: 100%|██████████| 36/36 [00:07<00:00,  4.72it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.98it/s]


                   all       1274       7377      0.862        0.6      0.677      0.437

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    100/100       4.7G     0.7033     0.3898     0.8574          6        640: 100%|██████████| 36/36 [00:07<00:00,  4.60it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.96it/s]


                   all       1274       7377      0.868      0.594      0.679      0.437

100 epochs completed in 0.382 hours.
Optimizer stripped from /home/sur06423/project/Version_5_exp/YOLO_11/v11_n/train/weights/last.pt, 5.5MB
Optimizer stripped from /home/sur06423/project/Version_5_exp/YOLO_11/v11_n/train/weights/best.pt, 5.5MB

Validating /home/sur06423/project/Version_5_exp/YOLO_11/v11_n/train/weights/best.pt...
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
                                                      CUDA:1 (NVIDIA RTX A6000, 48677MiB)
                                                      CUDA:2 (NVIDIA RTX A6000, 48677MiB)
                                                      CUDA:3 (NVIDIA RTX A6000, 48677MiB)
YOLO11n summary (fused): 238 layers, 2,582,932 parameters, 0 gradients, 6.3 GFLOPs


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:07<00:00,  2.73it/s]


                   all       1274       7377       0.87      0.597      0.678      0.438
            Pedestrian        333        600      0.804        0.6      0.677      0.382
         Traffic_Signs        309        434      0.899      0.492      0.591      0.336
               Vehicle       1034       2820      0.899       0.86      0.904      0.715
         traffic_light        660       3523      0.881      0.435      0.541      0.319
Speed: 0.1ms preprocess, 0.6ms inference, 0.0ms loss, 2.0ms postprocess per image
Results saved to /home/sur06423/project/Version_5_exp/YOLO_11/v11_n/train


# Results on Validation set:
100 epochs completed in 0.382 hours.
Optimizer stripped from /home/sur06423/project/Version_5_exp/YOLO_11/v11_n/train/weights/last.pt, 5.5MB
Optimizer stripped from /home/sur06423/project/Version_5_exp/YOLO_11/v11_n/train/weights/best.pt, 5.5MB

Validating /home/sur06423/project/Version_5_exp/YOLO_11/v11_n/train/weights/best.pt...
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
                                                      CUDA:1 (NVIDIA RTX A6000, 48677MiB)
                                                      CUDA:2 (NVIDIA RTX A6000, 48677MiB)
                                                      CUDA:3 (NVIDIA RTX A6000, 48677MiB)
YOLO11n summary (fused): 238 layers, 2,582,932 parameters, 0 gradients, 6.3 GFLOPs
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:07<00:00,  2.73it/s]
                   all       1274       7377       0.87      0.597      0.678      0.438
            Pedestrian        333        600      0.804        0.6      0.677      0.382
         Traffic_Signs        309        434      0.899      0.492      0.591      0.336
               Vehicle       1034       2820      0.899       0.86      0.904      0.715
         traffic_light        660       3523      0.881      0.435      0.541      0.319
Speed: 0.1ms preprocess, 0.6ms inference, 0.0ms loss, 2.0ms postprocess per image
Results saved to /home/sur06423/project/Version_5_exp/YOLO_11/v11_n/train

# Test the results on Test dataset
- The above results was on the validation set of the Carla_labeling-v5 dataset (See the Image counts)
- Now we need to calculate the metrics on the test set of the Carla_labeling-v5 dataset containing only 641 images.
- Below is the script to do so:

In [6]:
from ultralytics import YOLO

# Load YOLOv10s model from scratch
# model = YOLO("yolov10s.pt")
# Load YOLOv10s Best weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_11/v11_n/train/weights/last.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.25, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.6, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)


YOLO11n summary (fused): 238 layers, 2,582,932 parameters, 0 gradients, 6.3 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:10<00:00,  1.68s/it]


                   all        641       3701       0.84      0.593      0.748      0.527
            Pedestrian        182        332      0.776      0.596      0.733      0.455
         Traffic_Signs        168        230      0.823      0.504       0.69      0.449
               Vehicle        528       1377      0.906       0.85      0.913      0.782
         traffic_light        338       1762      0.854      0.421      0.656      0.424
Speed: 0.4ms preprocess, 1.5ms inference, 0.0ms loss, 2.1ms postprocess per image
Saving runs/detect/val/predictions.json...
Results saved to runs/detect/val


# Results on Last checpoint (100th epoch):
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
YOLO11n summary (fused): 238 layers, 2,582,932 parameters, 0 gradients, 6.3 GFLOPs
val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:10<00:00,  1.68s/it]
                   all        641       3701       0.84      0.593      0.748      0.527
            Pedestrian        182        332      0.776      0.596      0.733      0.455
         Traffic_Signs        168        230      0.823      0.504       0.69      0.449
               Vehicle        528       1377      0.906       0.85      0.913      0.782
         traffic_light        338       1762      0.854      0.421      0.656      0.424
Speed: 0.4ms preprocess, 1.5ms inference, 0.0ms loss, 2.1ms postprocess per image
Saving runs/detect/val/predictions.json...
Results saved to runs/detect/val

# Results on Test Set using Best Checkpoint:

In [7]:
from ultralytics import YOLO

# Load YOLOv10s model from scratch
# model = YOLO("yolov10s.pt")
# Load the model
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_11/v11_n/train/weights/best.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.25, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.6, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)


YOLO11n summary (fused): 238 layers, 2,582,932 parameters, 0 gradients, 6.3 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:09<00:00,  1.52s/it]


                   all        641       3701      0.858      0.586      0.749       0.53
            Pedestrian        182        332      0.821      0.578      0.734      0.459
         Traffic_Signs        168        230      0.839        0.5      0.693      0.452
               Vehicle        528       1377      0.912      0.847      0.912      0.781
         traffic_light        338       1762      0.859      0.421      0.658      0.428
Speed: 0.4ms preprocess, 1.1ms inference, 0.0ms loss, 2.3ms postprocess per image
Saving runs/detect/val2/predictions.json...
Results saved to runs/detect/val2


In [15]:
print(checkpoint.keys())

dict_keys(['date', 'version', 'license', 'docs', 'epoch', 'best_fitness', 'model', 'ema', 'updates', 'optimizer', 'train_args', 'train_metrics', 'train_results'])


# Taking the Optimal value from the F1-Confidence curve for Optimal Confidence
- Optimal Confidence YOLO11n - 0.28
- IoU threshold : 0.5
- Results save d at: runs/detect/val3

In [1]:
from ultralytics import YOLO

# Load YOLOv10s model from scratch
# model = YOLO("yolov10s.pt")
# Load YOLOv10s Best weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_11/v11_n/train/weights/last.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.28, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.5, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
YOLO11n summary (fused): 238 layers, 2,582,932 parameters, 0 gradients, 6.3 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:08<00:00,  1.35s/it]


                   all        641       3701      0.865       0.58      0.749      0.531
            Pedestrian        182        332        0.8      0.578      0.731      0.458
         Traffic_Signs        168        230      0.855      0.487      0.692      0.455
               Vehicle        528       1377      0.928      0.841      0.911      0.783
         traffic_light        338       1762      0.875      0.415       0.66      0.429
Speed: 1.9ms preprocess, 1.6ms inference, 0.0ms loss, 2.8ms postprocess per image
Saving runs/detect/val3/predictions.json...
Results saved to runs/detect/val3


# Best+ Taking the Optimal value from the F1-Confidence curve for Optimal Confidence
- Optimal Confidence YOLO11n - 0.28
- IoU threshold : 0.5
- Results save d at: runs/detect/val4

In [1]:
from ultralytics import YOLO

# Load YOLOv10s model from scratch
# model = YOLO("yolov10s.pt")
# Load YOLOv10s Best weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_11/v11_n/train/weights/best.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.28, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.5, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
YOLO11n summary (fused): 238 layers, 2,582,932 parameters, 0 gradients, 6.3 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:07<00:00,  1.28s/it]


                   all        641       3701      0.878      0.572      0.748      0.534
            Pedestrian        182        332      0.831      0.563      0.732      0.461
         Traffic_Signs        168        230      0.872      0.474      0.692      0.458
               Vehicle        528       1377      0.929      0.839       0.91      0.782
         traffic_light        338       1762       0.88      0.413      0.661      0.433
Speed: 1.9ms preprocess, 1.4ms inference, 0.0ms loss, 2.5ms postprocess per image
Saving runs/detect/val4/predictions.json...
Results saved to runs/detect/val4


# Best Checkpoint +  Optimal Confidence + IoU 0.6
- Optimal Confidence YOLO11n - 0.28
- IoU threshold : 0.6
- Results save d at: runs/detect/val4
- Best Epoch = 98

In [1]:
from ultralytics import YOLO

# Load YOLOv10s model from scratch
# model = YOLO("yolov10s.pt")
# Load YOLOv10s Best weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_11/v11_n/train/weights/best.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.28, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.6, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
YOLO11n summary (fused): 238 layers, 2,582,932 parameters, 0 gradients, 6.3 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:08<00:00,  1.35s/it]


                   all        641       3701      0.867      0.572      0.745      0.532
            Pedestrian        182        332       0.82      0.563      0.728      0.458
         Traffic_Signs        168        230      0.852      0.474      0.685      0.456
               Vehicle        528       1377       0.92       0.84       0.91      0.782
         traffic_light        338       1762      0.875      0.413      0.659      0.432
Speed: 1.9ms preprocess, 1.9ms inference, 0.0ms loss, 3.1ms postprocess per image
Saving runs/detect/val5/predictions.json...
Results saved to runs/detect/val5


# Seperate evaluation on the training set

In [1]:
from ultralytics import YOLO

# Load your model
model = YOLO('/home/sur06423/project/Version_5_exp/YOLO_11/v11_n/train/weights/best.pt')  # replace with your model path

# Validate on custom dataset
results = model.val(data='/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml',
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.25, # Sets the optimal confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.6, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="train", 
)  # Ensure your YAML points to your desired dataset

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
YOLO11n summary (fused): 238 layers, 2,582,932 parameters, 0 gradients, 6.3 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/train/labels.cache... 4485 images, 430 backgrounds, 0 corrupt: 100%|██████████| 4485/4485 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 36/36 [00:22<00:00,  1.57it/s]


                   all       4485      26788      0.884      0.617      0.777      0.573
            Pedestrian       1229       2227      0.841      0.652      0.785      0.538
         Traffic_Signs       1093       1459      0.882      0.526      0.727      0.481
               Vehicle       3636      10031      0.924      0.872      0.928      0.805
         traffic_light       2373      13071       0.89      0.418      0.669      0.468
Speed: 0.4ms preprocess, 1.1ms inference, 0.0ms loss, 0.7ms postprocess per image
Saving runs/detect/val6/predictions.json...
Results saved to runs/detect/val6


# This notebook is complete.
- All the cell are executed and saved for later use.
- Results are saved in the corresponding directories.